File: android/src/main/java/com/lwjlol/flutter_saveto/FlutterSavetoPlugin.java
package com.lwjlol.flutter_saveto;

import android.content.Intent;

import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.embedding.engine.plugins.activity.ActivityAware;
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
import io.flutter.plugin.common.PluginRegistry;

public class FlutterSavetoPlugin implements FlutterPlugin, ActivityAware {
    private ActivityPluginBinding activityPluginBinding;
    private HostApiIml hostApi;
    private final PluginRegistry.NewIntentListener onNewIntent = new PluginRegistry.NewIntentListener() {
        @Override
        public boolean onNewIntent(Intent intent) {
            return false;
        }
    };

    @Override
    public void onAttachedToEngine(FlutterPluginBinding binding) {
        hostApi = new HostApiIml(
                new FileSaver(
                        binding.getApplicationContext()
                )
        );
        Messages.SaveToHostApi.setUp(
                binding.getBinaryMessenger(),
                hostApi
        );
    }

    @Override
    public void onDetachedFromEngine(FlutterPluginBinding binding) {
        Messages.SaveToHostApi.setUp(binding.getBinaryMessenger(), null);
        if (hostApi != null) {
            hostApi.shutdown();
            hostApi = null;
        }
    }

    @Override
    public void onAttachedToActivity(ActivityPluginBinding binding) {
        activityPluginBinding = binding;
        activityPluginBinding.addOnNewIntentListener(onNewIntent);
    }

    @Override
    public void onDetachedFromActivityForConfigChanges() {
    }

    @Override
    public void onReattachedToActivityForConfigChanges(ActivityPluginBinding binding) {
    }

    @Override
    public void onDetachedFromActivity() {
        if (activityPluginBinding != null) {
            activityPluginBinding.removeOnNewIntentListener(onNewIntent);
        }
        activityPluginBinding = null;
    }
}

File: android/src/main/kotlin/com/lwjlol/flutter_saveto/FlutterSavetoPlugin.kt
package com.lwjlol.flutter_saveto

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.PluginRegistry


class FlutterSavetoPlugin : FlutterPlugin, ActivityAware {
    private var activityPluginBinding: ActivityPluginBinding? = null
    private val onNewIntent = PluginRegistry.NewIntentListener {
        return@NewIntentListener false
    }


    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        Messages.SaveToHostApi.setUp(
            binding.binaryMessenger,
            HostApiIml(FileSaver(binding.applicationContext)),
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    }


    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
        activityPluginBinding = binding
        activityPluginBinding?.addOnNewIntentListener(onNewIntent)
    }

    override fun onDetachedFromActivityForConfigChanges() {
    }

    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {

    }

    override fun onDetachedFromActivity() {
        activityPluginBinding?.removeOnNewIntentListener(onNewIntent);
        activityPluginBinding = null
    }
}